models/remoteworkitem.go

```diff
@@ -0,0 +1,77 @@
+package models
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// RemoteWorkItem represents a work item as it is stored in the database
+type RemoteWorkItem struct {
+	Fields map[string]interface{}
+}
+
+// WorkItemMap : holds the mapping between Remote key and local key in a WI
+type WorkItemMap map[string]interface{}
+
+// RemoteWorkItemSanitizer makes sure that the
+// Response from Provider API is sanitized into a
+// good looking flat 1-level dictionary
+type RemoteWorkItemMapper interface {
+	Flatten()
+	GetWorkItemKeyMap()
+}
+
+// GithubRemoteWorkItem is a derivative of RemoteWorkItem
+type GithubRemoteWorkItem struct {
```

Comment:
Thanks @aslakknutsen : 

I wanted to maintain the default key map at the ```GithubRemoteWorkItem``` implementation level and at the same time allow some flexibility to supply a different key maps at the ```MapRemoteWorkItem(..)```  invocation time, **also** because implementing the ```GetWorkItemKeyMap()``` method looks pretty optional. 

`var remoteWorkItemMaps = map[string]WorkItemMap{"github":  WorkItemMap{....}}`

I wouldn't want to do this because everytime we add  support for a new provider ( like Jira,bugzilla ), I wouldn't want a common data structure to be touched. ( unless we maintain a YAML/JSON file of maps )